packages/sdk/src/application/ApplicationClient.ts

```diff
@@ -438,31 +437,55 @@ export class ApplicationClient extends APIResource {
         await this._client.post<RestartPlaygroundConversationResponse>(
           queryString,
         );
-      return response;
+      return Promise.resolve(response);
     } catch (error) {
       return Promise.reject(error);
     }
   }
 
-  async chat(props: ChatRequest): Promise<ChatResponse> {
-    const { ownerId, appId, ...payload } = props;
+  async chat(
+    props: ChatRequest,
+    stream: boolean = true,
+  ): Promise<ReadableStream<Uint8Array>> {
+    const { ownerId, appId, requesterUid, accessToken, ...payload } = props;
+
+    if (
+      !accessToken ||
+      !ownerId ||
+      !appId ||
+      !payload.catalogId ||
+      !payload.conversationUid
+    ) {
+      throw new Error("Required parameters are missing");
+    }
 
-    try {
-      const queryString = getQueryString({
-        baseURL: `/namespaces/${ownerId}/apps/${appId}/chat`,
-      });
+    const queryString = getQueryString({
+      baseURL: `/namespaces/${ownerId}/apps/${appId}/chat`,
+    });
 
-      const additionalHeaders = getInstillAdditionalHeaders({
-        stream: true,
-      });
+    const additionalHeaders = getInstillAdditionalHeaders({
+      requesterUid,
+      stream,
+    });
 
-      const response = await this._client.post<ChatResponse>(queryString, {
-        body: JSON.stringify(payload),
-        additionalHeaders,
-      });
-      return response;
+    try {
+      const response = (await this._client.post(
+        queryString,
+        {
+          body: JSON.stringify(payload),
+          additionalHeaders,
+        },
+        /* eslint-disable @typescript-eslint/no-explicit-any */
+      )) as any;
+
+      if (stream) {
+        return response.body as ReadableStream<Uint8Array>;
+      } else {
+        return response.json();
+      }
     } catch (error) {
-      return Promise.reject(error);
+      console.error("Fetch failed:", error);
+      throw error;
```

Comment:
@thewbuk If you are using our class inherited method, it already handle streaming for you, all you need to do is directly return the response you get like this https://github.com/instill-ai/console/blob/109ff456d697af0f65f4c2fee5464554c33657cf/packages/sdk/src/vdp/trigger/TriggerClient.ts#L129